### PR TITLE
Add B field as an input for track parameter estimation

### PR DESCRIPTION
**Pull request by beomki-yeo**
Update: Instead of making an alternative one, I just make the original algorithm take the b field as an input argument

---------------------------
~~Adding a track parameter estimation with detector object which holds B field. I am not replacing the original one for faster development.~~

I am trying to reconstruct the momentum of spacepoints provided by @shimasnd and @stephenswat but current algorithm giving me a wrong value when I assumed 2 T magnetic field. I think there is a bug in the current algorithm but could you guys confirm the bfield strength?

### CODE REVIEW

**Comment by beomki-yeo** [comment]
Think I will just refactor the current algorithm without detector template

File: core/include/traccc/seeding/track_params_estimation_helper.hpp [hunk 1]
```diff
@@ -83,12 +83,13 @@ inline TRACCC_HOST_DEVICE bound_vector seed_to_bound_vector(
...
     // Curvature (with a sign) estimate
     scalar rho = -2.0f * B / getter::perp(vector2{1., A});
+
     // The projection of the top space point on the transverse plane of
     // the new frame
     scalar rn = local2[0] * local2[0] + local2[1] * local2[1];
     // The (1/tanTheta) of momentum in the new frame,
     scalar invTanTheta =
-        local2[2] * std::sqrt(1.f / rn) / (1.f + rho * rho * rn);
+        local2[2] * std::sqrt(1.f / rn) / (1.f + rho * rho * rn / 24.);
```
Comment (beomki-yeo) [inline core/include/traccc/seeding/track_params_estimation_helper.hpp:92]: Think I spotted a bug - without this term track_parameter_estimation test fails :p

**Comment by guilhermeAlmeida1** [comment]
I think in the end this should be set in the constructor of the algorithm with a `covfie::field_view` (which should come from `detector::get_bfield()` ). Definately not just a vector as we want to be able to handle non-const bfields.

**Comment by beomki-yeo** [comment]
OK I like that idea

File: tests/cpu/compare_with_acts_seeding.cpp [hunk 2]
```diff
@@ -70,10 +70,10 @@ inline bool operator==(const Acts::BoundVector& acts_vec,
             traccc::float_epsilon * 10 &&
         std::abs(acts_vec[Acts::eBoundTheta] -
                  traccc::getter::element(traccc_vec, traccc::e_bound_theta,
-                                         0)) < traccc::float_epsilon * 10 &&
+                                         0)) < traccc::float_epsilon * 1000 &&
```
Comment (beomki-yeo) [inline tests/cpu/compare_with_acts_seeding.cpp:73]: I had to mitigate the condition for equality operation on momentum direction as the specific bug is not fixed in ACTS yet.
I will revert the change once acts-project/acts#2211 is included in any of next releases

**Comment by beomki-yeo** [comment]
Can it be approved soon?

File: tests/cpu/compare_with_acts_seeding.cpp (hunk 2 above)
Comment (guilhermeAlmeida1) [reply tests/cpu/compare_with_acts_seeding.cpp:73]: This can be reverted now with https://github.com/acts-project/acts/releases/tag/v27.0.0

**Pull request merged**